lib/dotcom_web/live/trip_planner.ex

```diff
@@ -8,189 +8,354 @@ defmodule DotcomWeb.Live.TripPlanner do
-      |> assign(:form_values, AntiCorruptionLayer.convert_old_params(params))
-      |> assign(:map_config, @map_config)
-      |> assign(:from, [])
-      |> assign(:to, [])
-      |> assign(:submitted_values, nil)
-      |> assign_async(:results, fn ->
-        {:ok, %{results: nil}}
-      end)
-      |> assign(:itinerary_selection, :summary)
+      |> assign(@state)
+      |> assign(:input_form, Map.put(@state.input_form, :changeset, changeset))
+      |> maybe_submit_form()
 
-    {:ok, socket}
+    {:ok, new_socket}
   end
 
   @impl true
+  @doc """
+  The live view is made up of four subcomponents:
+
+  - InputForm: The form for the user to input their trip details
+  - ResultsSummary: A summary of the user's input and the number of results found
+  - Map: A map showing the user's trip details based on results and the selected itinerary group
+  - Results: Itinerary groups and itinerary details
+  """
   def render(assigns) do
     ~H"""
     <h1>Trip Planner <mark style="font-weight: 400">Preview</mark></h1>
     <div style="row">
+      <.input_form changeset={@input_form.changeset} />
+      <.results_summary changeset={@input_form.changeset} results={@results} />
       <.live_component
-        module={TripPlannerForm}
-        id={@form_name}
-        form_name={@form_name}
-        form_values={@form_values}
-      />
-      <section :if={@submitted_values} class="mt-2 mb-6">
-        <p class="text-lg font-semibold mb-0">{submission_summary(@submitted_values)}</p>
-        <p>{time_summary(@submitted_values)}</p>
-        <.async_result :let={results} assign={@results}>
-          <:failed :let={{:error, errors}}>
-            <.error_container title="Unable to plan your trip">
-              <p :for={%OpenTripPlannerClient.Error{message: message} <- errors} class="last:mb-0">
-                {message}
-              </p>
-            </.error_container>
-          </:failed>
-          <:loading>
-            <.spinner aria_label="Waiting for results" /> Waiting for results...
-          </:loading>
-          <%= if results do %>
-            <%= if Enum.count(results) == 0 do %>
-              <.feedback kind={:warning}>No trips found.</.feedback>
-            <% else %>
-              <.feedback kind={:success}>
-                Found {Enum.count(results)} {Inflex.inflect("way", Enum.count(results))} to go.
-              </.feedback>
-            <% end %>
-          <% end %>
-        </.async_result>
-      </section>
-
-      <.trip_planner_results_section
-        results={@results}
-        error={@error}
-        map_config={@map_config}
-        from={@from}
-        to={@to}
-        itinerary_selection={@itinerary_selection}
+        module={MbtaMetro.Live.Map}
+        id="trip-planner-map"
+        class="h-64 md:h-96 w-full"
+        config={@map.config}
+        lines={@map.lines}
+        pins={@map.pins}
+        points={@map.points}
       />
+      <.results results={@results} />
```

Comment:
The way that this is structured at the top level is (as far as I can tell) a regression, since now the map-plus-results-panel layout is wrong on both desktop and mobile. The map is supposed to get rendered to the side of the results panel on desktop, and on mobile, it's only supposed to be visible when viewing a specific itinerary (at which point it's rendered in between bits of the results panel).

I had originally introduced `TripPlannerResultsSection` (yeah, not the most elegant name 😇) in order to accomplish ☝️, and I'm struggling to see how I can avoid just re-doing that work.